src/mapper/map.rs

```diff
@@ -186,8 +187,45 @@ impl EventWriter {
         }
     }
 
-    pub fn to_native_script_event(&self) -> EventData {
-        EventData::NativeScript {}
+    pub fn to_native_script_event(&self, script: &PallasNativeScript) -> EventData {
+        let mut cbor = Vec::new();
+        minicbor::encode(script, &mut cbor).unwrap();
+        cbor.insert(0, 0);
+        let digest = Hasher::<224>::hash(&cbor);
```

Comment:
The policy ID is calculated in Oura because the `PallasNativeScript` struct is available with cbor serialization.

Doing it outside Oura, for example in a custom sink, would therefore require more work.

Reply:
LGTM. At some point, this should probably live as part of the Pallas primitives crate, that's how we've been handling hashes from other ledger-specific structs.

Reply:
As you wish. I can make a PR to Pallas instead and rework the patches on top of it. Your call.

Reply:
I say we move forward as currently implemented, we can refactor afterwards without breaking compatibility.